vcd/resource_vcd_nsxt_edgegateway_l2_vpn_tunnel.go

```diff
@@ -0,0 +1,399 @@
+				Optional: true,
+				Computed: true,
+				Description: "Network CIDR block over which the session interfaces. Only relevant if " +
+					"`session_mode` is set to `SERVER`",
+				ValidateFunc: validation.IsCIDR,
+			},
+			"connector_initiation_mode": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Description: "Connector initation mode of the session describing how a connection is made. " +
+					"Needs to be set only if `session_mode` is set to `SERVER`",
+				ValidateFunc: validation.StringInSlice([]string{"INITIATOR", "RESPOND_ONLY", "ON_DEMAND"}, false),
+			},
+			"pre_shared_key": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Description: "Pre-shared key used for authentication, needs to be provided only on creation of" +
+					"`SERVER` sessions",
+			},
+			"peer_code": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				Description: "Base64 encoded string of the full configuration of the tunnel, " +
+					"generated by the server-side L2 VPN session. This value contains the encoded pre-shared key " +
+					"so it is user's responsibility to secure it. Needs to be provided only on creation of" +
+					"`CLIENT` sessions",
+			},
+			"stretched_network": {
+				Type:     schema.TypeSet,
+				Optional: true,
+				// DHCP forwarding supports up to 8 IP addresses
+				Description: "Org VDC networks that are attached to this L2 VPN tunnel",
+				Elem:        stretchedNetwork,
+			},
+		},
+	}
+}
+
+var stretchedNetwork = &schema.Resource{
+	Schema: map[string]*schema.Schema{
+		"network_id": {
+			Type:        schema.TypeString,
+			Required:    true,
+			Description: "ID of the Org VDC network",
+		},
+		"tunnel_id": {
+			Type:         schema.TypeInt,
+			Optional:     true,
+			Computed:     true,
+			Description:  "Tunnel ID of the network for the tunnel. Read-only for `SERVER` sessions.",
+			ValidateFunc: validation.IntBetween(1, 4093),
+		},
+	},
+}
+
+func resourceVcdNsxtEdgegatewayL2VpnTunnelCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	vcdClient := meta.(*VCDClient)
+
+	unlock, err := vcdClient.lockParentVdcGroupOrEdgeGateway(d)
+	if err != nil {
+		return diag.Errorf("[L2 VPN Tunnel create] %s", err)
+	}
+
+	defer unlock()
+
+	orgName := d.Get("org").(string)
+	edgeGatewayId := d.Get("edge_gateway_id").(string)
+
+	nsxtEdge, err := vcdClient.GetNsxtEdgeGatewayById(orgName, edgeGatewayId)
+	if err != nil {
+		return diag.Errorf("[L2 VPN Tunnel create] error retrieving Edge Gateway: %s", err)
+	}
+
+	tunnelConfig, err := readL2VpnTunnelFromSchema(d)
+	if err != nil {
+		return diag.Errorf("[L2 VPN Tunnel create] %s", err)
+	}
+	tunnel, err := nsxtEdge.CreateL2VpnTunnel(tunnelConfig)
+	if err != nil {
+		return diag.Errorf("[L2 VPN Tunnel create] error creating L2 VPN Tunnel: %s", err)
+	}
+	d.SetId(tunnel.NsxtL2VpnTunnel.ID)
+
+	// As there may be warnings in the CreateUpdate function, we need to append them
```

Comment:
I don't see any appending happening here

Reply:
Removed it, yanked it from a different resource